batch.go

```diff
@@ -0,0 +1,126 @@
+package ed25519consensus
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/sha512"
+
+	"filippo.io/edwards25519"
+)
+
+// Verifier holds entries of public keys, signature and a scalar which are used for batch verification.
+type Verifier struct {
+	entries   []ks
+	batchSize uint32
+}
+
+// ks represents the public key, signature and scalar which the caller wants to batch verify
+type ks struct {
+	pubkey    ed25519.PublicKey
+	signature []byte
+	k         *edwards25519.Scalar
+}
+
+// NewVerifier creates a Verifier that entries of signatures, keys and messages
+// can be added to for verification
+func NewVerifier() Verifier {
```

Comment:
```suggestion
// NewBatchVerifier creates a BatchVerifier for batch verification.
func NewBatchVerifier() BatchVerifier {
```
same rationale as https://github.com/hdevalence/ed25519consensus/pull/7/files#r564128054 and https://github.com/hdevalence/ed25519consensus/pull/7/files#r564121892